src/D2L.CodeStyle.Analyzers/ApiUsage/ServiceLocator/SingletonLocatorAnalyzer.cs

```diff
@@ -88,96 +71,96 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
 			}
 		}
 
+		private delegate bool IsSingletonLocatorGet(
+			IMethodSymbol? methodSymbol
+		);
+
 		private delegate bool IsContainerType(
-				ITypeSymbol type,
-				[NotNullWhen( true )] out ITypeSymbol? containedType
-			);
+			ITypeSymbol type,
+			[NotNullWhen( true )] out ITypeSymbol? containedType
+		);
 
 		// Enforce that SingletonLocator can only load actual [Singleton]s
 		private static void EnforceSingletonsOnly(
-			SyntaxNodeAnalysisContext context,
-			Func<INamedTypeSymbol, bool> isSingletonLocator,
+			OperationAnalysisContext context,
+			IMethodSymbol methodSymbol,
+			IsSingletonLocatorGet isSingletonLocatorGet,
 			IsContainerType isContainerType
 		) {
-			var root = GetRootNode( context );
-			if( root == null ) {
+			if( !isSingletonLocatorGet( methodSymbol ) ) {
 				return;
 			}
-			var symbolinfo = context.SemanticModel.GetSymbolInfo( root, context.CancellationToken );
 
-			var method = symbolinfo.Symbol as IMethodSymbol;
+			ITypeSymbol typeArg = methodSymbol.TypeArguments.First();

```

Comment:
```suggestion
			ITypeSymbol typeArg = methodSymbol.TypeArguments.Single();
```

?

Reply:
Does this get called if the user has partially typed out `Get<>` or incorrectly written `Get<T, U>()`?

Reply:
Has to have compiled.